Sources/Views/MessageLabel.swift

```diff
@@ -22,417 +22,491 @@
+        
+        style.lineBreakMode = lineBreakMode
+        style.alignment = textAlignment
+
+        return style
+        
+    }
+    
     // MARK: - Initializers
-
+    
     public override init(frame: CGRect) {
+        
+        self.textContainer = NSTextContainer()
+        self.textContainer.lineFragmentPadding = lineFragmentPadding
+        self.textContainer.maximumNumberOfLines = numberOfLines
+        self.textContainer.lineBreakMode = lineBreakMode
+        self.textContainer.size = frame.size
+        
+        self.layoutManager = NSLayoutManager()
+        self.layoutManager.addTextContainer(self.textContainer)
+        
+        self.textStorage = NSTextStorage(attributedString: NSAttributedString(string: ""))
+        textStorage.addLayoutManager(layoutManager)
+        
         super.init(frame: frame)
-
-        // Message Label Specific
-        self.numberOfLines = 0
-        self.lineBreakMode = .byWordWrapping
-
+        
         let defaultAttributes: [NSAttributedStringKey: Any] = [
-          NSAttributedStringKey.foregroundColor: self.textColor,
-          NSAttributedStringKey.underlineStyle: NSUnderlineStyle.styleSingle.rawValue,
-          NSAttributedStringKey.underlineColor: self.textColor
+            .underlineStyle: NSUnderlineStyle.styleSingle.rawValue,
+            .underlineColor: UIColor.darkText
         ]
-
+        
         self.addressAttributes = defaultAttributes
+        self.phoneAttributes = defaultAttributes
         self.dateAttributes = defaultAttributes
-        self.phoneNumberAttributes = defaultAttributes
         self.urlAttributes = defaultAttributes
-
+        
         setupGestureRecognizers()
-
+        
+        backgroundColor = .clear
+        
     }
-
+    
     public required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
-    // MARK: - Open Methods
-
-    open override func drawText(in rect: CGRect) {
+    
+    // MARK: - Methods [Public]
+    
+    open override func draw(_ rect: CGRect) {
 
         let insetRect = UIEdgeInsetsInsetRect(rect, textInsets)
-        textContainer.size = CGSize(width: insetRect.width, height: rect.height)
+        textContainer.size = CGSize(width: insetRect.width, height: rect.height) // <- bug maybe?
 
-        let origin = insetRect.origin
         let range = layoutManager.glyphRange(for: textContainer)
+        let origin = insetRect.origin
 
         layoutManager.drawBackground(forGlyphRange: range, at: origin)
         layoutManager.drawGlyphs(forGlyphRange: range, at: origin)
+        
     }
-
-    // MARK: - Public Methods
-
-    // MARK: UIGestureRecognizer Delegate
-
-    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
-        return true
+    
+    public func configure(configurationBlock: () -> Void) {
```

Comment:
I think in the config block you should have a reference to message label and return Self.

```
@discardable_result
public func configure(configurationBlock: (MessageLabel) -> Void) -> Self
```

This could allow for clean inits like.
```
let label = MessageLabel().configure { label in
     // config here
}
```

Reply:
Possibly but right now it won't be used outside the cell